### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by andrei-at-luther**

### CODE REVIEW

File: substratecommon.go [hunk 1]
```diff
@@ -0,0 +1,815 @@
...
+	ErrorCode     int
+	ErrorMessage  string
+	ErrorJSON     []byte
+	TransactionID string
+}
+
+// Transaction represents summary information about a transaction.
+type Transaction struct {
+	ID     string
+	Reason string
```
Comment (sam-at-luther) [inline substratecommon.go]: These have been  updated to include the chaincode ID and the events. Please see latest APIs and make sure this is at parity.

File: substratecommon.go [hunk 2]
```diff
@@ -0,0 +1,817 @@
...
+}
+
+// RespCall encodes the response from Call
+type RespCall struct {
+	Response *Response
+	Err      *Error
+}
+
+// ArgsQueryInfo encodes the arguments to QueryInfo
+type ArgsQueryInfo struct {
```
Comment (sam-at-luther) [inline substratecommon.go:410]: Do all of these structs need to be exposed publically or are they just used internally?

File: substratecommon.go [hunk 3]
```diff
@@ -0,0 +1,817 @@
...
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/sirupsen/logrus"
+
+	hclog "github.com/hashicorp/go-hclog"
+)
+
+// ConcreteRequestOptions is a variant of RequestOptions that is
+// "flattened" to pure data.
+type ConcreteRequestOptions struct {
```
Comment (sam-at-luther) [inline substratecommon.go:21]: Do these request options need to be exposed publicly ?

File: substratecommon.go (hunk 2 above)
Comment (andrei-at-luther) [reply substratecommon.go:410]: Yes, I think RPC requires the fields to be public: `- the method has two arguments, both exported (or builtin) types.`

File: substratecommon.go (hunk 3 above)
Comment (andrei-at-luther) [reply substratecommon.go:21]: I guess yes for the same reason (it's passed through RPC).

**Pull request merged**